### PR TITLE
PHP8 CI

### PR DESCRIPTION
**Pull request by rullzer**
Good to go!

The test for other DBs will follow in later PRs

### CODE REVIEW

**Comment by MorrisJobke** [comment]
> [ ] Rebase once #23764 is in

Please also wait for #23834

**Comment by rullzer** [comment]
```
There was 1 failure:

1) Test\LargeFileHelperGetFileSizeTest::testGetFileSizeViaExec with data set #1 ('/drone/src/tests/data/strängé...2).txt', 446)
Failed asserting that null is identical to 446.

/drone/src/tests/lib/LargeFileHelperGetFileSizeTest.php:70
```

I think this is a locale issue...
See also: https://3v4l.org/vHqno

```
There was 1 error:

1) Test\Mail\EMailTemplateTest::testEMailTemplateSingleButton
ValueError: Unknown format specifier ";"

/drone/src/lib/private/Mail/EMailTemplate.php:592
/drone/src/tests/lib/Mail/EMailTemplateTest.php:180
```

No clue about that one yet.

**Comment by rullzer** [comment]
Ok lets review and see if we can still get this in.

File: apps/dav/tests/unit/BackgroundJob/UpdateCalendarResourcesRoomsBackgroundJobTest.php [hunk 1]
```diff
@@ -108,9 +112,9 @@ public function testRun() {
 		$backend3 = $this->createMock(IBackend::class);
 		$backend4 = $this->createMock(IBackend::class);
 
-		$res6 = $this->createMock([IResource::class, IMetadataProvider::class]);
```
Comment (nickvergessen) [inline apps/dav/tests/unit/BackgroundJob/UpdateCalendarResourcesRoomsBackgroundJobTest.php:111]: oh wait, that worked?

Comment (rullzer) [reply apps/dav/tests/unit/BackgroundJob/UpdateCalendarResourcesRoomsBackgroundJobTest.php:111]: Yes... But killed in php unit 9

**Comment by rullzer** [comment]
The psalm errors come from the fact that it doesn't support php8 properly i guess. since it doesn't know that openssl returns an object now. And the GD stuff a GdImage.

**Comment by rullzer** [comment]
Sure now oracle is complaining...

**Comment by nickvergessen** [comment]
Seems like https://github.com/nextcloud/server/blob/db82d27239aac99854415e6303e8802ce1dff2b2/lib/private/AppConfig.php#L134-L138 doesn't sort numbers before characters anymore

**Comment by ChristophWurst** [comment]
> The psalm errors come from the fact that it doesn't support php8 properly i guess. since it doesn't know that openssl returns an object now. And the GD stuff a GdImage.

could it be because of `build/stubs`?

**Comment by rullzer** [comment]
> > The psalm errors come from the fact that it doesn't support php8 properly i guess. since it doesn't know that openssl returns an object now. And the GD stuff a GdImage.
> 
> could it be because of `build/stubs`?

No clue why. But I chose to igrnoe it for now

**Review by ChristophWurst — APPROVED** [review]
I like the part where we skip tests -> faster CI :rocket:

**Review by nickvergessen — APPROVED** [review]
:shark: Bite by Bite

**Pull request merged**